File: backend/core/src/modules/plot-points/plot-points.controller.ts
import { Body, Controller, Delete, Get, Param, Patch, Post, Query, Req, UseGuards } from '@nestjs/common';
import { JwtAuthGuard } from '../auth/jwt-auth.guard';
import { PlotPointsService } from './plot-points.service';

@UseGuards(JwtAuthGuard)
@Controller('plot-points')
export class PlotPointsController {
  constructor(private readonly plotPointsService: PlotPointsService) {}

  @Get()
  findAll(@Req() req: any, @Query('plotId') plotId?: string) {
    if (!plotId) return [];
    return this.plotPointsService.findAllForUser(req.user?.userId, plotId);
  }

  @Get(':id')
  findOne(@Req() req: any, @Param('id') id: string) {
    return this.plotPointsService.findOneForUser(req.user?.userId, id);
  }

  @Post()
  create(@Req() req: any, @Body() dto: any) {
    return this.plotPointsService.createForUser(req.user?.userId, dto);
  }

  @Patch(':id')
  update(@Req() req: any, @Param('id') id: string, @Body() dto: any) {
    return this.plotPointsService.updateForUser(req.user?.userId, id, dto);
  }

  @Delete(':id')
  async remove(@Req() req: any, @Param('id') id: string) {
    await this.plotPointsService.removeForUser(req.user?.userId, id);
    return { success: true };
  }
}

File: backend/core/src/main/java/store/pjcloud/cowrite/core/controller/PlotPointsController.java
package store.pjcloud.cowrite.core.controller;

import java.util.List;
import java.util.Map;
import java.util.UUID;
import org.springframework.web.bind.annotation.*;
import store.pjcloud.cowrite.core.entity.PlotPoint;
import store.pjcloud.cowrite.core.security.SecurityUtils;
import store.pjcloud.cowrite.core.service.PlotPointsService;

@RestController
@RequestMapping("/plot-points")
public class PlotPointsController {
    private final PlotPointsService service;

    public PlotPointsController(PlotPointsService service) {
        this.service = service;
    }

    @GetMapping
    public List<PlotPoint> list(@RequestParam("plotId") UUID plotId) {
        UUID userId = SecurityUtils.requireUserId();
        return service.findAllForUser(userId, plotId);
    }

    @GetMapping("/{id}")
    public PlotPoint get(@PathVariable("id") UUID id) {
        UUID userId = SecurityUtils.requireUserId();
        return service.findOneForUser(userId, id);
    }

    @PostMapping
    public PlotPoint create(@RequestBody PlotPoint body) {
        UUID userId = SecurityUtils.requireUserId();
        return service.createForUser(userId, body);
    }

    @PatchMapping("/{id}")
    public PlotPoint update(@PathVariable("id") UUID id, @RequestBody PlotPoint body) {
        UUID userId = SecurityUtils.requireUserId();
        return service.updateForUser(userId, id, body);
    }

    @DeleteMapping("/{id}")
    public Map<String, Object> delete(@PathVariable("id") UUID id) {
        UUID userId = SecurityUtils.requireUserId();
        service.removeForUser(userId, id);
        return Map.of("success", true);
    }
}
